.github/workflows/ci.yml

```diff
@@ -24,7 +22,13 @@ jobs:
       # and hyphens.
       run: echo "TAG=branch-${GITHUB_REF_NAME//[^A-Za-z0-9._-]/-}" | tee -a $GITHUB_ENV
 
-    - run: ./devel/tag $TAG
+    - if: ${{ github.event_name != 'pull_request' }}
+      uses: docker/login-action@v1
+      with:
+        username: ${{ secrets.DOCKER_LOGIN }}
+        password: ${{ secrets.DOCKER_PASSWORD }}
+
+    - run: ./devel/build-tag-push $TAG
```

Comment:
This needs more careful thinking. Now, the PR run fails on `build-tag-push` even though it's just the "push" part that won't work (build is fine). Push should be skipped on PR runs.

Reply:
This is addressed by 3ad8608cd8da0e833c482f3e27f2d77552c976ad + fixups.